Copyedit why-i-joined-temporal-charles-zedlewski (#884)
temporalio/documentation: blog/why-i-joined-temporal-charles-zedlewski.md

For nearly my entire career, I’ve worked in B2B software; I can’t imagine doing anything else.
Software is an innately creative medium.
If you learn a craft and find the right place to apply it, you can manifest change with your own two hands and play a positive and meaningful role in the march of progress.

I’ve been especially lucky to get to play a role in two category-defining companies that have each left an impact larger than their revenues or market caps.
For 5 years I worked at SAP, relatively late in its life, and then for another 10 years at Cloudera, starting in the very early days before the first shipped product.
Temporal is a marriage of the domains of both companies: transactional applications and distributed open source platforms.

Transactional and business process applications have been the backbone of the global economy for as long as there’s been software.
All of the world’s materials, assets, locations, people, customers, money, and obligations arrive, move, evolve, and depart at the behest of these kinds of applications.

> Many of these are packaged applications written by SAP and its successors and even more of them are bespoke applications written by engineering teams working in companies of all shapes and sizes.

Reviewer: ```suggestion
Many of these are packaged applications written by SAP and its successors, and even more of them are bespoke applications written by engineering teams working in companies of all shapes and sizes.
```

In the past 15 years, the infrastructure for how we run these applications has changed dramatically, and this has transformed how applications are built and architected.
Cloud infrastructure—like containers, kubernetes, and PaaS—have made it easy and inexpensive to build applications from an ever greater number of discrete components.
These cloud architectures have produced all sorts of benefits.  Ever-larger engineering teams can now develop & enhance applications with greater agility.

> Scaling has become much simpler and less expensive.  Reuse is increasingly common, with developers borrowing from neighboring applications or from commercial API’s that allow teams to efficiently outsource things like messaging, identity, or payment processing.

Reviewer: ```suggestion
Scaling has become much simpler and less expensive.
Reuse is increasingly common, with developers borrowing from neighboring applications or from commercial APIs that allow teams to efficiently outsource things like messaging, identity, or payment processing.
```

Temporal upends all of this.
Temporal is a new kind of platform. It’s not a database, cache, queue, or a means to run your code.
It’s a runtime for distributed applications that transparently manages their state at any scope and scale.
By externalizing state to a generalized platform like Temporal, application teams get to outsource the majority of cross-cutting concerns presented by cloud applications.
In doing so, applications become:

- **More reliable to run:** Temporal applications fail to execute less often, and when parts of the application do fail, they always recover to a consistent state
- **More productive to enhance:** It’s not uncommon to see examples of engineers implementing a feature in Temporal with 40-60% fewer lines of code than before, as they're outsourcing most of the distributed application concerns to Temporal.
- **Easier to operate:** Temporal helps users see how their production application executes its logic, consolidates errors, and provides simple but powerful mechanisms to make fixes without interrupting the operation of the running application.

> At write time, the Temporal SDK provides engineers a simple set of primitives native to their preferred language, the most central of which are Workflows.

Reviewer: Perhaps change "At write time" to "During development"? (Like Loren, I found that this word choice caused a short stumble.)

Cloudera used open source and no small amount of inspiration from Google to make scale a ubiquitous commodity.
When I joined Cloudera in 2010, “petabyte scale” was synonymous with “exotic gear.”
So exotic that the leading database players referred to petabyte scale organizations as a “club” whose membership list was small enough to fit on a single sheet of paper.
Cloudera lowered the cost of scale by 2 orders of magnitude, and today there are many thousands of companies who run at petabyte scale.
This went on to have a large & positive impact on the world.
Markets and businesses were created, new revenue and cost savings were unearthed, customer careers were made, and sometimes even lives were saved.

In a similar way, **Temporal is using open source and distributed systems to make reliability a ubiquitous commodity at any scale**, and this will have an even more substantial impact than commoditizing scale.
Lowering the human and technical cost of reliability and reparability by orders of magnitude means we will come to expect these characteristics more often and in more places.

> Reliability as a commodity will give engineers freedom to take risks, as they can now move fast and not break things.

Reviewer: ```suggestion
Reliability as a commodity will give engineers freedom to take risks, because they can now move fast and not break things.
```
